using err in string interpolation gives compile error

**V version: 0.1.17** 
**OS: MacOS** 

**What did you do?**
printing an error inside an `or` block
```
user := repo.find_user_by_id(99) or { // Option types must be handled by `or` blocks  
		// println(err) // OK
		println('error message:$err') // NOK
		return  // `or` block must end with `return`, `break`, or `continue`  
	} 
```

**What did you expect to see?**
error message: [contents of err]
 
**What did you see instead?**
.option_result.c:3211:7: error: passing 'User' (aka 'struct User') to parameter of incompatible type 'void *'
 free(user); // close_scope free
      ^~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/malloc/_malloc.h:42:18: note: passing argument to parameter here
void     free(void *);
                    ^
1 error generated.
V panic: clang error

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -1029,7 +1029,7 @@ fn (p mut Parser) close_scope() {
 			else if v.typ == 'string' { 
 				p.genln('v_string_free($v.name); // close_scope free') 
 			} 
-			else { 
+			else if v.ptr {
 				p.genln('free($v.name); // close_scope free') 
 			} 
 		} 

